src/konnektor/tests/network_planners/generators/test_explicit_network_planner.py

```diff
@@ -28,33 +29,42 @@ def test_explicit_network_planner():
     assert get_is_connected(ligand_network)
 
 
-def test_explicit_network_planner_with_indices():
+def test_explicit_network_planner_from_indices():
+    pass
+
+
+def test_explicit_network_planner_from_indices_disconnected():
     n_compounds = 20
     components, genMapper, genScorer = build_random_dataset(n_compounds=n_compounds)
     indices = [(1, 2), (2, 3), (3, 4), (2, 5), (2, 6)]
-    unique_indices = set([i for e in indices for i in e])
     planner = ExplicitNetworkGenerator(genMapper, genScorer, n_processes=1)
 
-    ligand_network = planner.generate_network_from_indices(components=components, indices=indices)
+    with pytest.warns(match="Generated network is not fully connected"):
+        ligand_network = planner.generate_network_from_indices(
+            components=components, indices=indices
+        )
 
     assert isinstance(ligand_network, LigandNetwork)
-    assert len(ligand_network.nodes) == len(unique_indices)
+    assert len(ligand_network.nodes) == n_compounds
     assert len(ligand_network.edges) == len(indices)
```

Comment:
For here and below, it would be good to explicitlly test that the edges are composed oof the same indices pairs.

Reply:
agreed, done.